app/code/Magento/Ui/Controller/Adminhtml/Index/Render.php

```diff
@@ -30,6 +30,7 @@ public function execute()
         $component = $this->factory->create($this->_request->getParam('namespace'));
         $this->prepareComponent($component);
         $this->_response->appendBody((string) $component->render());
+        $this->_response->setHeader('Content-Type', 'application/json', true);
```

Comment:
Main problem with hard-coding `Content-Type` here is that the `$component->render()` has multiple types of renderers. 

You may refer to `\Magento\Ui\Component\AbstractComponent::render` which in place selects the render engine from the classes which implement `\Magento\Framework\View\Element\UiComponent\ContentType\ContentTypeInterface`.

This list is not limited to Json, but also has an Html and Xml implementations out of the box.
I suggest looking into different approach to solve this issue, such as: 
* Possibly limiting the `Content-Type` header to the correct renderer
* Working with the `Accept` header provided by client to ensure the correct response.